src/certmgr/cert_mgmt.c

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (c) 2023 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <mbedtls/x509_crt.h>
+#include <psa/protected_storage.h>
+#include "sntp_client.h"
+#include "dhcpwait.h"
+#include "bootstrap.h"
+#include "provision.h"
+#include "key_mgmt.h"
+#include "util_app_log.h"
+
+LOG_MODULE_REGISTER(cert_mgmt, CONFIG_LOG_DEFAULT_LEVEL);
+
+#define CERT_MGMT_THREAD_PRIORITY 7
+
+#define TEN_AND_HALF_DAYS_IN_SEC 907200
```

Comment:
This should be a kconfig value, ideally, although we can default to 10.5 days for the value